### PR TITLE
Flatten MCP server config: remove content_roots, hoist kind/mounts

### PR DESCRIPTION
**Pull request by dylanturn**
## Summary

Simplifies the MCP server configuration data model from a two-level hierarchy (server → content_roots → mounts) to a flat structure (server → mounts). The `kind` field (simple/virtual) moves from content_roots onto the server itself, eliminating the intermediate table and reducing API/UI complexity.

## Key Changes

**Backend (Python)**
- Removed `McpServerContentRoot` model and `mcp_server_content_roots` table
- Added `kind` column to `McpServer` (simple/virtual, defaults to simple)
- Repointed `McpServerMount.content_root_id` → `McpServerMount.mcp_server_id`
- Updated validation in `_validate_mounts()` to enforce:
  - Simple servers: exactly one mount with empty `virtual_prefix`
  - Virtual servers: one or more mounts with non-overlapping prefixes
- Simplified API request/response models: `ContentRootInput` removed, `McpServerCreate`/`McpServerUpdate` now take `kind` + `mounts` directly
- Updated all route handlers, resolver, and test fixtures to use flat structure
- Added Alembic migration (0004) that:
  - Copies `kind` from first content_root to server
  - Repoints all mounts to their server
  - Drops non-first content_roots (single-tenant dev feature, no production data)
  - Provides downgrade path for reversibility

**Frontend (React/TypeScript)**
- Removed `ContentRoot` interface; `McpServer` now has `kind` and `mounts` directly
- Refactored `CreateMcpServerModal` to manage `kind` and `mounts` as top-level state
- Replaced nested `ContentRootEditor` with flat `MountEditor` component
- Added `KindToggle` UI for switching between simple/advanced modes
- Simplified validation and error handling (field errors now use `_mounts` instead of `_content_roots`)
- Updated `McpServersTab` to count mounts directly

**Tests**
- Updated all test fixtures to use flat structure
- Renamed test cases (e.g., `test_create_with_simple_root` → `test_create_with_simple_mount`)
- Simplified test payloads and assertions

## Implementation Details

- The migration preserves the first content_root's kind and mounts for each server; additional roots are discarded (this was a UI-unsupported edge case in development)
- Simple servers enforce empty `virtual_prefix` at the database level via validation
- The UI now uses a toggle to switch between simple (single mount) and advanced (multiple mounts with prefixes) modes
- All error messages and field names updated to reflect the new structure (`_mounts` instead of `_content_roots`)

https://claude.ai/code/session_01FwfHApqXWWk8KwcLusoe3J

### CODE REVIEW

File: stash_mcp/tenant_admin/mcp_servers.py [hunk 1]
```diff
@@ -284,55 +266,44 @@ def _validate_runtime_compatibility(
...
+    """Per-server validation: kind matches mount count, mounts have
+    normalized paths, no prefix overlap.
+
+    Zero mounts is allowed at the type level (an unmounted server
+    exposes nothing — the runtime resolver refuses it with a clear
+    error). ``simple`` requires exactly one mount with an empty
+    ``virtual_prefix``; ``virtual`` requires at least one and no
+    overlapping prefixes.
+    """
+    if not mounts:
```
[the remark this answers is not in the document]
Comment (dylanturn) [reply stash_mcp/tenant_admin/mcp_servers.py:287]: Good catch — fixed in cb9d824. `_validate_mounts` now rejects `kind='virtual'` with empty mounts before the early return, with tests covering both the create and patch paths.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: alembic/versions/0004_drop_content_roots.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    # Recreate one content root per server, carrying its kind.
+    bind = op.get_bind()
+    bind.execute(
+        sa.text(
+            """
+            INSERT INTO mcp_server_content_roots
+                (id, mcp_server_id, name, description, kind, sort_order)
+            SELECT
+                lower(hex(randomblob(16))), id, slug, description, kind, 0
+              FROM mcp_servers
```
[the remark this answers is not in the document]
Comment (dylanturn) [reply alembic/versions/0004_drop_content_roots.py]: Fixed in cb9d824. Replaced the raw `randomblob(16)`/`hex()` SQL with a Python-side `uuid.uuid4()` bound through a `sa.table` abstraction, so SQLAlchemy's `Uuid` type handles the dialect-specific binding for both SQLite and PostgreSQL.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: alembic/versions/0004_drop_content_roots.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+    # Recreate one content root per server, carrying its kind.
+    bind = op.get_bind()
+    bind.execute(
+        sa.text(
+            """
+            INSERT INTO mcp_server_content_roots
+                (id, mcp_server_id, name, description, kind, sort_order)
+            SELECT
+                lower(hex(randomblob(16))), id, slug, description, kind, 0
```
[the remark this answers is not in the document]
Comment (dylanturn) [reply alembic/versions/0004_drop_content_roots.py]: Duplicate of the comment above — addressed in cb9d824.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: stash_mcp/tenant_admin/mcp_servers.py [hunk 4]
```diff
@@ -284,55 +266,44 @@ def _validate_runtime_compatibility(
...
+) -> list[MountInput]:
+    """Per-server validation: kind matches mount count, mounts have
+    normalized paths, no prefix overlap.
+
+    Zero mounts is allowed at the type level (an unmounted server
+    exposes nothing — the runtime resolver refuses it with a clear
+    error). ``simple`` requires exactly one mount with an empty
+    ``virtual_prefix``; ``virtual`` requires at least one and no
+    overlapping prefixes.
+    """
```
[the remark this answers is not in the document]
Comment (dylanturn) [reply stash_mcp/tenant_admin/mcp_servers.py:282]: Duplicate of the earlier comment — addressed in cb9d824.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: stash_mcp/tenant_admin/mcp_servers.py [hunk 5]
```diff
@@ -662,68 +612,93 @@ async def update_mcp_server(
...
+                    subpath=m.subpath,
+                    virtual_prefix=m.virtual_prefix,
+                )
+                for m in server.mounts
+            ]
+        )
+        # When only kind changes, source_mounts above has empty slugs;
+        # resolve them from the existing rows so validation has real
+        # store_slugs to look up.
+        if body.mounts is None:
```
[the remark this answers is not in the document]
Comment (dylanturn) [reply stash_mcp/tenant_admin/mcp_servers.py]: Good catch — fixed in a3b3388. `MountInput.store_slug` enforces `min_length=1`, so the placeholder construction raised a Pydantic `ValidationError` before the resolve-stores block could fill in real slugs (every kind-only PATCH would 422). Refactored to resolve the existing stores first and build `MountInput`s with real slugs in one pass. Added a regression test (`test_patch_kind_only_rehydrates_existing_mounts`) that exercises both an incoherent kind flip and a coherent one.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: stash_mcp/tenant_admin/mcp_servers.py [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
+Configs live inert until spec 03 wires tokens to them and spec 04 turns
+them on at runtime.
 
 Validation rules enforced here (defense-in-depth — the UI also
 prevents these):
 
-- ``simple`` content roots must have exactly one mount; ``virtual`` must
-  have at least one.
+- ``kind='simple'`` servers must have exactly one mount with empty
+  ``virtual_prefix``; ``kind='virtual'`` must have at least one.
```
[the remark this answers is not in the document]
Comment (dylanturn) [reply stash_mcp/tenant_admin/mcp_servers.py]: Right — fixed in a3b3388. The module docstring now reads "at most one mount" for `kind='simple'`, matching what `_validate_mounts` actually enforces (zero is allowed and means the server is inert; the runtime resolver refuses unmounted servers with a clear error).

---
_Generated by [Claude Code](https://claude.ai/code)_
